monty/dev.py

```diff
@@ -39,18 +58,23 @@ def craft_message(old, replacement, message):
             else:
                 r = replacement
             msg += f"; use {r.__name__} in {r.__module__} instead."
-        if message is not None:
+
+        if message:
             msg += "\n" + message
         return msg
 
-    def deprecated_decorator(old):
+    def deprecated_decorator(old: Callable):
         def wrapped(*args, **kwargs):
-            msg = craft_message(old, replacement, message)
+            msg = craft_message(old, replacement, message, deadline)
             warnings.warn(msg, category=category, stacklevel=2)
             return old(*args, **kwargs)
 
         return wrapped
 
+    # Raise a CI error after removal deadline
+    if deadline is not None and "CI" in os.environ and datetime.now() > deadline:
```

Comment:
maybe better to use `os.getenv("CI)` in case it's set to False (should only ever be set to `True` or be absent but just in case)

```suggestion
    if deadline and os.getenv("CI") and datetime.now() > deadline:
```

Reply:
Thanks for reviewing and the suggestions. I decide maybe it's better to take an int tuple instead of `datetime` for maximum user (developer) convenience. Also I noticed the docs need to be updated too.

Reply:
I didn't change `if deadline is not None` to `if deadline` here as we didn't implement a strict type check...

From [PEP-8](https://peps.python.org/pep-0008/):
>Comparisons to singletons like None should always be done with is or is not, never the equality operators.
>Also, beware of writing if x when you really mean if x is not None – e.g. when testing whether a variable or argument that defaults to None was set to some other value. The other value might have a type (such as a container) that could be false in a boolean context!